### DISCUSSION TITLE
Setting a caption for a table in conjunction with a bookmark

### DISCUSSION THREAD

**Discussion by ReneHei** [davidgohel/flextable#552] [Q&A]
Hello,

I have an issue what I can not solve on my own. Maybe someone has an idea to help me.

Issue: I can´t set a caption for a table in conjunction with a bookmark. (It works for charts for example: read_docx("some Word with bookmarks_graph") %>% docx_bookmarks()   %>% cursor_bookmark( "NameOfBookmark")%>%body_add_par ("SomeTitle") %>% body_add_gg(Waterfall) ).

**But this is not working for tables:**

read_docx("some Word with bookmarks") %>% 
docx_bookmarks()   %>%
 cursor_bookmark( "NameOfBookmark")%>%  body_add_par("My flextable",  style = "heading 3")%>%
          body_add_flextable (value = table, pos = "on") %>% 
                                     print(target = "U/...")

Thank you very much and I hope its sounds not to trivial.

**Comment by davidgohel** [comment]
> `body_add_flextable (value = table, pos = "on")`

If you add the table where is the cursor, you replace the content then the bookmark is lost because the flextable is now the new content

**Reply by ReneHei** [reply]
thank you David, 

it finally worked, when I wrote the "pos ="on" " inside the  body_add_par().

Thanks a lot.

:)

**Comment by ReneHei** [comment]
Hi David,


Thank you very much.

I still don’t understand it.

May I ask one more question: Why is body_add_flextable() working if I add the table to the bookmark without the caption?

I tried to add the caption during the creation of the table and it works if I only print(…docx), so without the bookmark.

With the bookmark I get an error something like "the the style is not matching” - I tried out the different styles

Bw

René

**Reply by davidgohel** [reply]
I can't help with no reproducible example. As always and forever.

**Reply by davidgohel** [reply]
are you sure your question is not covered in the documentation? https://ardata-fr.github.io/flextable-book/captions-and-cross-references.html or https://davidgohel.github.io/flextable/reference/set_caption.html? If not (which is possible) you need to provide a reproducible example, something I can copy paste and update or comment so that we clearly are looking at the same thing
